Modules/_decimal/_decimal.c

```diff
@@ -2350,7 +2350,16 @@ dec_from_long(decimal_state *state, PyTypeObject *type, PyObject *v,
         PyLong_FreeExport(&export_long);
     }
     else {
-        mpd_qset_i64(MPD(dec), export_long.value, ctx, status);
+        const int64_t value = export_long.value;
+
+        if (INT32_MIN <= value && value <= INT32_MAX) {
```

Comment:
Is it possible to accept UINT32_MAX?

```suggestion
        if (-UINT32_MAX <= value && value <= UINT32_MAX) {
```